src/Tools/dotnet-monitor/ServiceCollectionExtensions.cs

```diff
@@ -13,6 +13,14 @@ namespace Microsoft.Diagnostics.Tools.Monitor
 {
     internal static class ServiceCollectionExtensions
     {
+        public static IServiceCollection ConfigureMetrics(this IServiceCollection services, IConfiguration configuration)
+        {
+            services.AddSingleton((IOptionsChangeTokenSource<MetricsOptions>)new ConfigurationChangeTokenSource<MetricsOptions>(configuration.GetSection(MetricsOptions.ConfigurationKey)));
```

Comment:
This should not be necessary. The below call to configure the options from configuration should be handling changes automatically.